### PR TITLE
Add support for DELETE request

### PR DESCRIPTION
**Pull request by mikeurbach**
@grobie please review

it would be handy for my apps if the DELETE api was exposed by the ruby client

### CODE REVIEW

File: lib/prometheus/client/push.rb [hunk 1]
```diff
@@ -36,6 +36,10 @@ def replace(registry)
         request('PUT', registry)
       end
 
+      def delete
+        request('DELETE')
```
Comment (grobie) [inline lib/prometheus/client/push.rb]: Given we're not going to send the registry, I guess we shouldn't pretend that in the Content-Type header. Wdyt about just calling `@http.send_request(...)` directly here?

Comment (mikeurbach) [inline lib/prometheus/client/push.rb]: makes sense to me, i will update it

**Comment by grobie** [comment]
Thanks, LGTM. Please add a small stub to the readme example as well.

**Comment by mikeurbach** [comment]
@grobie i just realized i accidentally checked in a version of the gem i had built for testing... do you want me to make a PR or will you remove it?

**Comment by grobie** [comment]
I saw that and understood it as an accident, but it doesn't really hurt
either. I'll bump it correctly once I release the new version, don't worry
about it.

**Comment by mikeurbach** [comment]
oh, ya i was uploading this to a private gem server for testing. but i mean i actually checked in the built gem to the codebase: https://github.com/prometheus/client_ruby/blob/master/prometheus-client-0.7.0.beta.gem

**Comment by grobie** [comment]
Oooh, shoot. Thanks for the heads-up, I removed it.

**Pull request merged**